test/e2e/network_segmentation.go

```diff
@@ -905,6 +1219,117 @@ func normalizeConditions(conditions []metav1.Condition) []metav1.Condition {
+		BlockOwnerDeletion: pointer.Bool(true),
+		Controller:         pointer.Bool(true),
+	}}))
+	ExpectWithOffset(1, nad.Labels).To(Equal(map[string]string{"k8s.ovn.org/user-defined-network": ""}))
+	ExpectWithOffset(1, nad.Finalizers).To(Equal([]string{"k8s.ovn.org/user-defined-network-protection"}))
+
+	expectedNetworkName := "cluster.udn." + udnName
+	expectedNadName := namespace + "/" + udnName
+	ExpectWithOffset(1, nad.Spec.Config).To(MatchJSON(`{
+		"cniVersion":"1.0.0",
+		"type": "ovn-k8s-cni-overlay",
+		"name": "` + expectedNetworkName + `",
+		"netAttachDefName": "` + expectedNadName + `",
+		"topology": "layer2",
+		"role": "secondary",
+		"subnets": "10.100.0.0/16"
+	}`))
+}
+
+func assertClusterUDNStatusReportsActiveNamespaces(cudnName string, expectedActiveNsNames ...string) {
+	conditionsRaw, err := e2ekubectl.RunKubectl("", "get", "clusteruserdefinednetwork", cudnName, "-o", "jsonpath='{.status.conditions}'")
+	ExpectWithOffset(1, err).NotTo(HaveOccurred())
+	conditionsRaw = strings.ReplaceAll(conditionsRaw, `\`, ``)
+	conditionsRaw = strings.ReplaceAll(conditionsRaw, `'`, ``)
+	var conditions []metav1.Condition
+	ExpectWithOffset(1, json.Unmarshal([]byte(conditionsRaw), &conditions)).To(Succeed())
+
+	c := conditions[0]
+	// equality matcher cannot be used since condition message namespaces order is inconsistent
+	ExpectWithOffset(1, c.Type).Should(Equal("NetworkReady"))
+	ExpectWithOffset(1, c.Status).Should(Equal(metav1.ConditionTrue))
+	ExpectWithOffset(1, c.Reason).Should(Equal("NetworkAttachmentDefinitionReady"))
+
+	ExpectWithOffset(1, c.Message).To(ContainSubstring("NetworkAttachmentDefinition has been created in following namespaces:"))
+	for _, ns := range expectedActiveNsNames {
+		Expect(c.Message).To(ContainSubstring(ns))
+	}
+}
+
+func assertClusterUDNStatusReportConsumers(conditionsJSON, udnName, udnNamespace, expectedPodName string) {
+	conditionsJSON = strings.ReplaceAll(conditionsJSON, `\`, ``)
+	conditionsJSON = strings.ReplaceAll(conditionsJSON, `'`, ``)
+
+	var conditions []metav1.Condition
+	ExpectWithOffset(1, json.Unmarshal([]byte(conditionsJSON), &conditions)).To(Succeed())
+	conditions = normalizeConditions(conditions)
+	expectedMsg := fmt.Sprintf("failed to delete NetworkAttachmentDefinition [%[1]s/%[2]s]: network in use by the following pods: [%[1]s/%[3]s]",
+		udnNamespace, udnName, expectedPodName)
+	ExpectWithOffset(1, conditions).To(Equal([]metav1.Condition{
+		{
+			Type:    "NetworkReady",
+			Status:  "False",
+			Reason:  "NetworkAttachmentDefinitionSyncError",
+			Message: expectedMsg,
+		},
+	}))
+}
+
+func newClusterUDNManifest(name string, targetNamespaces ...string) string {
+	targetNs := strings.Join(targetNamespaces, ",")
+	return `
+apiVersion: k8s.ovn.org/v1
+kind: ClusterUserDefinedNetwork
+metadata:
+  name: ` + name + `
+spec:
+  namespaceSelector:
+    matchExpressions:
+    - key: kubernetes.io/metadata.name
+      operator: In
+      values: [ ` + targetNs + ` ]
```

Comment:
I tried to switch to having common label for all but it doesnt seem to it easier.

In most tests there  is no need to update both CUDN and namespace to  select new namespace, it require changing CUDN because test use the  `kubernetes.io/metadata.name` label.

Reply:
I think we should make a copy for "namespace-selector is mutated" test, to use non-namespace-name label and than mutate namespace labels instead of selector and also check that it is handled correctly.
Original problem was that we didn't watch for namespace changes correctly, so to test this case, we need to change namespace labels only and not touch a CUDN itself.

Reply:
I think its covered in unit-tests by the test "when existing namespace's matching label removed, should delete stale NADs in previously matching namespaces". 
Does it make senses to add e2e for it as well?